Remotion/Web/ClientScript/Scripts/SmartNavigation.ts

```diff
@@ -0,0 +1,249 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+class SmartScrolling_Element
+{
+  constructor (
+    public readonly ID: string, 
+    public readonly Top: number, 
+    public readonly Left: number)
+  {
+  }
+
+  public ToString()
+  {
+    if (StringUtility.IsNullOrEmpty (this.ID))
+      return '';
+    else
+      return this.ID + ' ' + this.Top + ' ' + this.Left;
+  };
+
+  public static Parse (value: Nullable<string>): Nullable<SmartScrolling_Element>
+  {
+    ArgumentUtility.CheckTypeIsString ('value', value);
+    if (StringUtility.IsNullOrEmpty (value))
+      return null;
+  
+    var fields = value.split (' ');
+    // TODO: fields is string[] but second and third parameter expect a number
+    return new SmartScrolling_Element (fields[0], fields[1] as unknown as number, fields[2] as unknown as number);
+  };
+}
+
+class SmartScrolling
+{
+  public static Restore (data: Nullable<string>): void
+  {
+    ArgumentUtility.CheckTypeIsString ('data', data);
+    if (StringUtility.IsNullOrEmpty (data))
+      return;
+          
+    var dataFields = data.split ('*');
+    if (dataFields.length == 0)
+      return;
+    
+    var dataField = dataFields[0];
+    dataFields = dataFields.slice (1);
+    var sseBody = SmartScrolling_Element.Parse (dataField);
+    // TODO: sseBody can be null
```

Comment:
Intersting. apprently that's never the case or there'd be an exception. Bug and TBD which way the fix should go.